docs/GHAFileExplained.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: page
+title: GitHub Actions CI workflow file explained
+---
+
+Below is the
+[gha.dist.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/master/gha.dist.yml)
+file with comments added to explain what each section is doing. For additional
+information please refer to [workflow syntax reference](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions).
+
+If you are familiar with Travis, it should be straightforward to understand
+the new syntax, also you may find this [migration
+manual](https://docs.github.com/en/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions)
+useful.
+
+```yaml
+# Title of the workflow
+name: Moodle Plugin CI
+
+# Run this workflow every time a new commit pushed to your repository or PR
+# created.
+on: [push, pull_request]
+
+jobs:
+  # Set the job key. The key is displayed as the job name
+  # when a job name is not provided
+  test:
+    # Virtual environment to use.
+    runs-on: ubuntu-18.04
+
+    # DB services you need for testing.
+    services:
+      postgres:
+        image: postgres:9.6
+        env:
+          POSTGRES_USER: 'postgres'
+          POSTGRES_HOST_AUTH_METHOD: 'trust'
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 3
+      mariadb:
+        image: mariadb:10
+        env:
+          MYSQL_USER: 'root'
+          MYSQL_ALLOW_EMPTY_PASSWORD: "true"
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval 10s --health-timeout 5s --health-retries 3
+
+    # Determines build matrix. This is a list of PHP versions, databases and
+    # branches to test our project against. For each combination a separate
+    # build will be created. For example below 6 builds will be created in
+    # total (7.2-pgsql, 7.2-mariadb, 7.3-pgsql, 7.3-mariadb, etc.). If we add
+    # another branch, total number of builds will become 12.
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ['7.2', '7.3', '7.4']
```

Comment:
Didn't we comment about to add a note for people wanting to use php70 and coverage, recommending them to downgrade to ubuntu 16? My memory fails!

Reply:
Ah, good idea.